forte/mrdsrg-spin-integrated/dsrg_mrpt2_grad/dsrg_mrpt2_deriv_tensors.cc

```diff
@@ -0,0 +1,150 @@
+/**
+ * Set CI-related and DSRG tensors.
+ */
+#include "psi4/libpsi4util/PsiOutStream.h"
+#include "../dsrg_mrpt2.h"
+#include "helpers/timer.h"
+
+using namespace ambit;
+using namespace psi;
+
+namespace forte {
+
+void DSRG_MRPT2::set_ci_ints() {
+    ci = ci_vectors_[0];
+    // X_K <K|p^+ q|0> + <0|p^+ q|K> X_K
+    Gamma1_tilde = BTF_->build(CoreTensor, "Gamma1_tilde", spin_cases({"aa"}));
+    // X_K <K|p^+ q^+ r s|0> + <0|p^+ q^+ r s|K> X_K
+    Gamma2_tilde = BTF_->build(CoreTensor, "Gamma2_tilde", spin_cases({"aaaa"}));
+}
+
+void DSRG_MRPT2::set_density() {
+    Gamma2_ = BTF_->build(CoreTensor, "Gamma2_", spin_cases({"aaaa"}));
```

Comment:
I am wondering: do we need to make a copy of the 2RDM? I think that `g2aa()` and the other functions can give access to RDMs without copying data (please verify).

Reply:
I think the`RDMs` calss now returns a copy, so I think this is OK.